File: Day1/ArmStrong.py
#function 1 to get the number of digit in the number for the power
def get_digit_count(num):
    count = 0
    temp = num
    while temp>0:
        count += 1
        temp //=  10
    return count
#function 2 to get the power of the digit
def power_of(digit,power):  
    prod=1 
    while power>0:
        prod *= digit
        power -= 1
    return prod
#function 3 to check the number is ArmStrong or not
def is_armstrong_number(num):
    temp=num
    total_sum = 0
    power=get_digit_count(num)
    while temp>0:
        digit = temp % 10
        total_sum +=  power_of(digit,power)
        temp //= 10
    return num == total_sum

num = int(input("Enter the number : "))
ans = is_armstrong_number(num)
if ans:
    print(num,"is a Arm Strong Number. ")
else:
    print(num,"is not a Arm Strong Number.")

File: Day1/ArmStrong.cpp
#include<iostream>
using namespace std;

int getDigitCount(int num){
    int count = 0;
    while(num > 0){
        count++;
        num /= 10;
    }
    return count;
}
int getPower(int digit,int power){
    int prod =1;
    while(power>0){
        prod *= digit;
        power--;
    }
    return prod;
}
bool getIsArmStrongNumber(int num){
    int temp = num;
    int sum = 0;
    int power = getDigitCount(num);
    while( num>0){
        int digit = num % 10;
        sum  += getPower(digit,power);
        num /= 10;
    }
    return temp == sum;
}
int main(){
    int num;
    cout<<"Enter the Number : ";
    cin>>num;
    bool ans = getIsArmStrongNumber(num);
    if (ans == true)
    {
       cout<<num<<" is a Arm Strong Number";
    }
    else{
        cout<<num<<" is not a Arm Strong Number.";
    }
    return 0;
}